### PR TITLE
Sockets: update tests for Linux 6.6 fix.

### LINKED ISSUE #94149: Linux Kernel Regression on tcp socket disconnection track issue
This issue originally started with [this commit](https://github.com/torvalds/linux/commit/4faeee0cf8a5d88d63cdbc3bab124fb0e6aed08c) on the Linux kernel.
For more info see: #91543

There is a patch that is written by a kernel dev: [Link](https://marc.info/?l=linux-netdev&m=169686209917378&w=4)
Let's see if that patch will get included, or if we need to find another solution to this problem.

Related issues to this: 
#93735
#93737

/cc: @tmds

### PR DESCRIPTION
**Pull request by tmds**
Fixes https://github.com/dotnet/runtime/issues/94149.

@antonfirsov @liveans ptal.

### CODE REVIEW

**Review by liveans — APPROVED** [review]
LGTM, but do we know if this patch (original behavior) will be backported to all supported kernel versions? If yes, I believe we need to wait until CI machines to be updated with the new patch and we need to update PR to activate this test for `all` kernel versions.

Also, we need to backport this to 8.0, 7.0 and 6.0.

Please correct me if I'm wrong @antonfirsov.

/cc @karelz

**Comment by tmds** [comment]
> but do we know if this patch (original behavior) will be backported to all supported kernel versions? If yes, I believe we need to wait until CI machines to be updated with the new patch and we need to update PR to activate this test for all kernel versions.

It's not expected for this change to be backported to earlier kernels.

> Also, we need to backport this to 8.0, 7.0 and 6.0.

You may backport it if you want to assert the stricter behavior also on the release branches.
Checking it on the main branch may also be good enough.

**Review by antonfirsov — APPROVED** [review]
Thanks!

**Pull request merged**
